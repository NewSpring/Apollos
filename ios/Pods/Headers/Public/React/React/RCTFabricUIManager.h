../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h