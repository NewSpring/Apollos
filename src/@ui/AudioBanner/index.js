import React from 'react';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import styled from '@ui/styled';
import PaddedView from '@ui/PaddedView';
import Icon from '@ui/Icon';
import { H7 } from '@ui/typography';

const enhance = compose(
  pure,
  setPropTypes({
    titleText: PropTypes.string,
  }),
  defaultProps({
    titleText: 'Listen to audio',
  }),
);

const Banner = styled({
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
})(PaddedView);

const AudioBanner = enhance(({ titleText }) => (
  <Banner>
    <View>
      <H7>{titleText}</H7>
    </View>
    <Icon name={'audio'} />
  </Banner>
));

export default AudioBanner;
