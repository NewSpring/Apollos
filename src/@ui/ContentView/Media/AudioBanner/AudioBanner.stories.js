import React from 'react';
import { storiesOf } from '@storybook/react-native';

import AudioBanner from './';

storiesOf('AudioBanner', module)
  .add('Default', () => <AudioBanner />)
  .add('Custom title', () => <AudioBanner titleText={'Listen to Sermon'} />);
