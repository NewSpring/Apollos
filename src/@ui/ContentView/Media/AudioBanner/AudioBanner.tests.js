import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';
import AudioBanner from './';

describe('the AudioBanner component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <AudioBanner />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept a title prop', () => {
    const tree = renderer.create(
      <Providers>
        <AudioBanner titleText={'Listen to Sermon'} />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
