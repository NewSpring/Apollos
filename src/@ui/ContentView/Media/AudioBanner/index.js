import React from 'react';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import ConnectedImage from '@ui/ConnectedImage';
import { withTheme } from '@ui/theme';
import { withMediaPlayerActions } from '@data/mediaPlayer';
import styled from '@ui/styled';
import Touchable from '@ui/Touchable';
import PaddedView from '@ui/PaddedView';
import Icon from '@ui/Icon';
import { H7 } from '@ui/typography';

const enhance = compose(
  pure,
  setPropTypes({
    mediaId: PropTypes.string,
    titleText: PropTypes.string,
    currentTrack: PropTypes.shape({
      title: PropTypes.string,
      file: PropTypes.string,
      duration: PropTypes.string,
    }),
    playlist: PropTypes.shape({
      title: PropTypes.string.isRequired,
      images: ConnectedImage.propTypes.source.isRequired,
      colors: PropTypes.arrayOf(PropTypes.shape({ value: PropTypes.string })).isRequired,
    }),
  }),
  defaultProps({
    titleText: 'Listen to audio',
  }),
  withTheme(({ theme: { helpers: { rem = {} } = {} } = {} }) => ({ rem })),
  withMediaPlayerActions,
);

const Banner = styled(({ theme }) => ({
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
  backgroundColor: theme.colors.lightTertiary, // TODO: move color into theme type
}))(PaddedView);

const AudioBanner = enhance(
  ({
    mediaId, setNowPlaying, currentTrack, playlist, titleText, rem,
  }) => (
    <Touchable onPress={() => setNowPlaying({ id: mediaId, currentTrack, playlist })}>
      <Banner>
        <View>
          <H7>{titleText}</H7>
        </View>
        <Icon name={'audio'} size={rem(1)} />
      </Banner>
    </Touchable>
  ),
);

export default AudioBanner;
