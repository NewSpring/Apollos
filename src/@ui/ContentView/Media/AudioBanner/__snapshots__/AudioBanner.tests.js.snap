// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the AudioBanner component should accept a title prop 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#dddddd",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Regular",
            "fontSize": 14,
            "lineHeight": 15.78,
            "paddingBottom": 1.8,
            "paddingTop": 3.6,
          }
        }
      >
        Listen to Sermon
      </Text>
    </View>
    <Svg
      height={18}
      viewBox="0 0 24 24"
      width={18}
    >
      <Path
        d="M3.82 13.5H2v-.32C2 6.54 5.22 2 12 2c6.5 0 9.64 4.16 9.98 10.42v1.07h-1.82v-1c-.3-5.8-3.02-8.8-8.16-8.8-5.37 0-8.18 3.32-8.18 9.45v.3zm0 1.7h1.6c.87 0 1.58.66 1.58 1.48v2.13c0 .86-.7 1.5-1.6 1.5-.87 0-1.58-.64-1.58-1.5v-3.6zm0-1.7h1.6c1.87 0 3.4 1.42 3.4 3.18v2.13c0 1.8-1.53 3.2-3.4 3.2C3.5 22 2 20.6 2 18.8v-5.3h1.82zm16.36 1.7v3.6c0 .83-.7 1.5-1.6 1.5-.87 0-1.58-.67-1.58-1.5v-2.1c0-.82.7-1.5 1.6-1.5h1.58zm0-1.7H22v5.3c0 1.77-1.53 3.2-3.4 3.2-1.9 0-3.42-1.43-3.42-3.2v-2.1c0-1.76 1.53-3.2 3.4-3.2h1.6z"
        fill="#303030"
      />
    </Svg>
  </View>
</View>
`;

exports[`the AudioBanner component should render 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#dddddd",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Regular",
            "fontSize": 14,
            "lineHeight": 15.78,
            "paddingBottom": 1.8,
            "paddingTop": 3.6,
          }
        }
      >
        Listen to audio
      </Text>
    </View>
    <Svg
      height={18}
      viewBox="0 0 24 24"
      width={18}
    >
      <Path
        d="M3.82 13.5H2v-.32C2 6.54 5.22 2 12 2c6.5 0 9.64 4.16 9.98 10.42v1.07h-1.82v-1c-.3-5.8-3.02-8.8-8.16-8.8-5.37 0-8.18 3.32-8.18 9.45v.3zm0 1.7h1.6c.87 0 1.58.66 1.58 1.48v2.13c0 .86-.7 1.5-1.6 1.5-.87 0-1.58-.64-1.58-1.5v-3.6zm0-1.7h1.6c1.87 0 3.4 1.42 3.4 3.18v2.13c0 1.8-1.53 3.2-3.4 3.2C3.5 22 2 20.6 2 18.8v-5.3h1.82zm16.36 1.7v3.6c0 .83-.7 1.5-1.6 1.5-.87 0-1.58-.67-1.58-1.5v-2.1c0-.82.7-1.5 1.6-1.5h1.58zm0-1.7H22v5.3c0 1.77-1.53 3.2-3.4 3.2-1.9 0-3.42-1.43-3.42-3.2v-2.1c0-1.76 1.53-3.2 3.4-3.2h1.6z"
        fill="#303030"
      />
    </Svg>
  </View>
</View>
`;
